### DISCUSSION TITLE
Feedback: UI

### DISCUSSION THREAD

**Discussion by 6543** [woodpecker-ci/woodpecker#2162] [General]
Summary of UI issues and smaller changes which can be posted as comments.
For complex improvements please open a separate issue.  
*Followup of  #468, witch helped quite a lot already :)*

# UI improvements
- [ ] add some explore page (similar to github / gitea), showing a list of public repos (could be home page for not authenticated users instead of login screen)
- [x] improve pagination
- [x] add a label/tag `disabled` on /repos view for disabled repos (https://github.com/woodpecker-ci/woodpecker/issues/468#issuecomment-991672016)
- [x] on disabled repos -> repo config -> repo actions: deactivate button should be hidden (https://github.com/woodpecker-ci/woodpecker/issues/468#issuecomment-991672389)
- [x] on disabled repos -> repo config -> repo actions: add activate button

**Comment by 6543** [comment]
- [x] horizontal scroll-bar for  config view:
![Bildschirmfoto zu 2022-01-16 18-40-38](https://user-images.githubusercontent.com/24977596/149671381-54bc120a-c969-42cc-a01d-85458ee81724.png)

----
https://github.com/woodpecker-ci/woodpecker/issues/468#issuecomment-1013920101

**Comment by 6543** [comment]
@eladyn:

In WebKit (Safari, GNOME Web, ...) the animated circle indicator is set off from the correct location, doesn't happen in Chromium or Firefox.
![woodpecker_ui_bug](https://user-images.githubusercontent.com/59307989/151639310-94f87ef8-19a0-4165-acdb-4b1b71b39f08.png)

---
https://github.com/woodpecker-ci/woodpecker/issues/468#issuecomment-1024770819

**Comment by 6543** [comment]
@valentinleistner:

It would be nice if you could see the current build status and maybe last commit and branch on the repos page

<img width="1322" alt="screenshot of the current repos page" src="https://user-images.githubusercontent.com/77448526/152203360-3007bf8e-40c4-4287-8680-9f9222e7a211.png">

---
https://github.com/woodpecker-ci/woodpecker/issues/468#issuecomment-1028165170

---

See #1872

**Comment by 6543** [comment]
**anbraten**: Pipelines in branches tab (for example master) also include Pull-requests against master => should not contain PRs
> **6543**: I would say they should, you can target PRs to different branches
> > **6543**:  ... we might also wana add some more fiters so you can toggle this

---
https://github.com/woodpecker-ci/woodpecker/issues/468#issuecomment-1029521524, https://github.com/woodpecker-ci/woodpecker/issues/468#issuecomment-1234466143

**Comment by 6543** [comment]
@qwerty287:

It would be nice to see public projects of other users on their "profile" list.

---
https://github.com/woodpecker-ci/woodpecker/issues/468#issuecomment-1108620680

**Reply by qwerty287** [reply]
-> https://github.com/woodpecker-ci/woodpecker/issues/2983

**Comment by 6543** [comment]
![image](https://user-images.githubusercontent.com/24977596/197307313-21514c44-5421-4879-9042-dda4c82a8385.png)

move matrix lable in same column as name cc @lukashass

**Reply by xoxys** [reply]
Don't know if that works, as there is the time for the pipeline displayed now:

<img src="https://github.com/woodpecker-ci/woodpecker/assets/3391958/3214b096-ad29-402b-ba4c-374de0116143" width="250">

Any other idea? Or just keep it in its own column?

**Reply by 6543** [reply]
hmm well it's tru if the lable is long it currently works out quite nicely

**Comment by 6543** [comment]
pipelines ~and job steps~ with long names look out of place

![image](https://user-images.githubusercontent.com/24977596/197357846-7052df23-843b-4e9d-adcb-43d32a1aa276.png)

---
source: https://codeberg.org/Codeberg-CI/feedback/issues/85

**Comment by qwerty287** [comment]
> https://github.com/woodpecker-ci/woodpecker/issues/1314#issuecomment-1287392140

This is implemented already.

**Comment by anbraten** [comment]
![image](https://user-images.githubusercontent.com/6918444/198874007-97e3f0ed-2097-4d78-a926-b0a2d26492b9.png)

Instead of `/` in front of `Settings` a `-` would separate it a bit better I guess.

**Comment by lafriks** [comment]
Slash `/` is more common for this imho

**Comment by anbraten** [comment]
![image](https://user-images.githubusercontent.com/6918444/198891651-05ac58b8-66a5-483b-b95c-216ddb49ea59.png)

**Comment by gapodo** [comment]
Just seen in the Woodpecker-CI (the projects CI)

![image](https://user-images.githubusercontent.com/49525042/199854398-6be5eea3-7ceb-403b-99d3-d6393b2a350e.png)
https://ci.woodpecker-ci.org/woodpecker-ci/woodpecker/pipeline/4431 (screenshot done on a 2560 wide screen without zoom / scaling)

It seems this is not consistent since https://ci.woodpecker-ci.org/woodpecker-ci/woodpecker/pipeline/4429 does not show this overflowing outside of the window and should be based on the same PR (the screenshot is the merge onto master, while the working one is the PR).

**Comment by anbraten** [comment]
![image](https://user-images.githubusercontent.com/6918444/202860216-532a46e2-0bdd-438a-9b7f-535dc0c37133.png)

Missing margin for pipelines with matrix and single workflow

**Comment by 6543** [comment]
If a pipeline needs aproval, it does not show a link back to the forge, so you manually have to figure out what the pull is :(

so it encureage to just aprove and look afterwards what was the change ... !!!!!!

**Reply by xoxys** [reply]
Addressed by https://github.com/woodpecker-ci/woodpecker/pull/2345

**Comment by anbraten** [comment]
@ivictbor Looks like you are using `0.15` there should be already quite some improvements of it in the `next` version.

**Comment by ivictbor** [comment]
> @ivictbor Looks like you are using `0.15` there should be already quite some improvements of it in the `next` version.

Thanks, yes, exactly, now 0.15.5, always updating once new version is available

**Comment by 6543** [comment]
![2022-12-07_15-39](https://user-images.githubusercontent.com/24977596/207061208-714c65ff-099f-49f9-90b9-06c2be473fb0.png)

hide after some length and make a mose hover over detail dialog

**Comment by MFTabriz** [comment]
> ![2022-12-07_15-39](https://user-images.githubusercontent.com/24977596/207061208-714c65ff-099f-49f9-90b9-06c2be473fb0.png)

also, in case of a long commit message, whole message (and ←) should stay in 1 line and not wrap into 3. On small displays this is very important.

**Comment by anbraten** [comment]
![image](https://user-images.githubusercontent.com/6918444/222882209-460d5f41-9b3a-4e81-bd9a-1f65947ab714.png)

use proper icon size

**Comment by maxkratz** [comment]
Possible improvement: Add freely configurable links to the top left navigation bar. With this "feature", the admins are able to, for example, add a link to the VCS like Forgejo/Gitea/...
Using the Forgejo/Gitea config option, one can already add a link "CI" to the Woodpecker instance. (Thus it would be nice to add the link back to the VCS.)

**Comment by dR3b** [comment]
The fonts are way too big (console output).

**Comment by laurentS** [comment]
The job output is unreadable for long lines:
![image](https://user-images.githubusercontent.com/1745184/224188849-175d314d-5043-41bf-8182-17fe3e840593.png)

The above screenshot is on a 1980px width window. I've confirmed by looking at the page source that the full text is there, it just needs wrapping before the right hand sidebar, not under it :) This is `v0.15.6`.

**Comment by anbraten** [comment]
@laurentS This could be fixed in [next](https://woodpecker-ci.org/faq#which-version-of-woodpecker-should-i-use) already. You can also close the right sidepanel by clicking on the `0` next to your avatar:
![image](https://user-images.githubusercontent.com/6918444/224255218-d8174ab5-adbf-4547-b0a0-aa811cb72387.png)

**Comment by laurentS** [comment]
@anbraten I've updated to `next`, thanks for pointing this out! Closing the side panel does the job indeed, so I guess that it's no big deal if it covers the logs when open. Thanks for your help!

**Comment by lafriks** [comment]
Imho executing job list should be moved from side menu to it's own page

**Comment by anbraten** [comment]
![image](https://user-images.githubusercontent.com/6918444/226159111-76a0ad5d-2251-4fe1-ad4c-6df5d1717b7a.png)

**Comment by 6543** [comment]
[source](https://matrix.to/#/!AiponSMjDBfbRjzkZn:obermui.de/$fpzsolwhNufPs6dlUYZqmq3oJc5cL_IjNhtCg1S7QtE?via=obermui.de&via=matrix.org&via=kle.li)

" ... copying lines from the log is painful since it also copies the line number and duration on either side of the log ... "

**Comment by lafriks** [comment]
> [source](https://matrix.to/#/!AiponSMjDBfbRjzkZn:obermui.de/$fpzsolwhNufPs6dlUYZqmq3oJc5cL_IjNhtCg1S7QtE?via=obermui.de&via=matrix.org&via=kle.li)
> 
> " ... copying lines from the log is painful since it also copies the line number and duration on either side of the log ... "

Seems to be working correctly for me

**Comment by qwerty287** [comment]
Yes, that seems to be an old version (maybe the one with xterm.js?).

**Comment by qwerty287** [comment]
UI isn't updated after declining pipeline

**Reply by anbraten** [reply]
fixed meanwhile

**Comment by anbraten** [comment]
![image](https://user-images.githubusercontent.com/6918444/226504832-87a94c98-d310-4dc4-96a6-eaa03c7cc75e.png)

Match icon and stats colors

**Comment by lafriks** [comment]
> ![image](https://user-images.githubusercontent.com/6918444/226504832-87a94c98-d310-4dc4-96a6-eaa03c7cc75e.png)
> 
> Match icon and stats colors

Stat colors are a bit differently based -> green (free resources), blue (used resources), red (over resources) while for task status colors mean a bit different things

**Comment by lafriks** [comment]
![attels](https://user-images.githubusercontent.com/165205/226623763-26ea69b4-453f-4702-bf44-f0b5dd37958f.png)
- [ ] 1) maybe move status to some other place (after or before title)?
- [x] 2) do not show labels that has no valuable information

**Comment by 6543** [comment]
On Admin Settings, User View: if you create a user you wont get back to user list view on save (on first klick)

**Comment by MFTabriz** [comment]
In matrix builds, name of the variable (here, "SOMETHING") should be vertically centered.
![image](https://github.com/woodpecker-ci/woodpecker/assets/39205857/a318c30b-1af5-456b-9bf4-74991aa1cb0f)

**Reply by xoxys** [reply]
https://github.com/woodpecker-ci/woodpecker/pull/4497

**Comment by 6543** [comment]
Collor Schema Change/Fix: #2003

**Comment by 6543** [comment]
> The fonts are way too big (console output).

user setting: font size
?

**Comment by dR3b** [comment]
@6543 In the following screenshots, I will show you my suggestion for improvements:

1) ~~The font is way too big. On the left, as mainly that of the console.~~
2) Why not use the space on the left so that the name is not cut off? -> https://github.com/woodpecker-ci/woodpecker/discussions/2162#discussioncomment-6728032
3) ~~An icon for the status of the entire job would be useful here.~~
4) ~~Always show the log file download dialog.~~

Take a look at the screenshot of Drone. The overall picture is more coherent.

![20230725-082257](https://github.com/woodpecker-ci/woodpecker/assets/9670938/bdd913ed-ce39-407e-9a07-b60019cdba35)
![20230725-082350](https://github.com/woodpecker-ci/woodpecker/assets/9670938/2c134d5a-ea21-45cd-984b-c2d67de006dc)

Thanks!

**Comment by xoxys** [comment]
1. (At least for the console logs) and 3. are included in https://github.com/woodpecker-ci/woodpecker/pull/2003

**Comment by ambroisie** [comment]
Could we get the current status (i.e: latest workflow status) for each project on the homepage?

**Comment by qwerty287** [comment]
The pipeline counter has a quadrat background but a circle foreground

![grafik](https://github.com/woodpecker-ci/woodpecker/assets/80460567/ebd09442-69ea-41ff-91fc-ea015d158b13)

There should be some margin between square brackground border and circle foreground, just like it is for buttons with isLoading

**Comment by qwerty287** [comment]
@ambroisie see #1872

**Comment by 6543** [comment]
![image](https://github.com/woodpecker-ci/woodpecker/assets/24977596/14a27e1f-7bc1-42be-bb24-6e24601b7780)

make it klickable and go to pipeline list filtered by branch e.g.: https://ci.woodpecker-ci.org/repos/3780/branches/release%2Fv1.0 in this case

**Comment by xoxys** [comment]
> The pipeline counter has a quadrat background but a circle foreground
> 
> ![grafik](https://user-images.githubusercontent.com/80460567/257554372-ebd09442-69ea-41ff-91fc-ea015d158b13.png)

@qwerty287 Am I blind :sweat_smile: ? The background and hover are squared, just the spinner is round, or what do you mean?

**Comment by qwerty287** [comment]
@xoxys yes, I mean the spinner. Would be nice if it would spin around the background element as it was before with a round background

**Comment by xoxys** [comment]
Hmm :thinking:  personally I don't like the idea and don't think this would look nice.

**Comment by qwerty287** [comment]
Do you think the current one looks good? At least, there should be some margin between the spinner and the borders of the background element. I think it would be better to completely use rounded squares.

**Comment by xoxys** [comment]
We also use circle spinners in buttons etc. so yes I think if we want to have something spinning it should be a circle, but that might just be a personal preference.

**Comment by qwerty287** [comment]
Hmm yes, from this side you're right. But there the spinners have the margin :upside_down_face: (so yes that would work too)

**Comment by xoxys** [comment]
The Pipeline Log view should show repo name:
![image](https://github.com/woodpecker-ci/woodpecker/assets/3391958/b2a6d408-3179-44fd-8101-cdedf45c63bd)

**Reply by dR3b** [reply]
Yes please and use all the free space:
Original:
```
.max-w-5xl {

    max-width: 64rem;

}
```
Fix:
```
.max-w-5xl {

    max-width: 99%;

}
```
![20230815-093719](https://github.com/woodpecker-ci/woodpecker/assets/9670938/3310bbc6-d311-4a97-83a3-d7ed5b21546d)

**Reply by xoxys** [reply]
Addressed by https://github.com/woodpecker-ci/woodpecker/pull/2327

**Comment by eliasscosta** [comment]
Could we have the name of the repository here?  When we share the link from a pipeline, sometimes it is difficult to understand which the repo is. 
![image](https://github.com/woodpecker-ci/woodpecker/assets/22245125/3dde7ac8-d73f-49ce-aeb7-c85bb5592702)

**Reply by xoxys** [reply]
That's already done on `next`.

![image](https://github.com/woodpecker-ci/woodpecker/assets/3391958/dc265b94-a257-4ccb-b93c-f29dd9f04680)

**Comment by qwerty287** [comment]
Make tabs `<a>` links to be able to e.g. open in new browser tabs

**Reply by anbraten** [reply]
https://github.com/woodpecker-ci/woodpecker/issues/3827

**Comment by anbraten** [comment]
Favicons are showing a failed pipeline when navigating to its page and its still running. When reloading the pipeline view the correct favicon is used.

**Comment by qwerty287** [comment]
From matrix/@6543:

Woodpecker running icon is cut off:
![grafik](https://github.com/woodpecker-ci/woodpecker/assets/80460567/4406d5ff-5228-4b8d-9213-2e1ac0971003)

**Comment by 6543** [comment]
BUG: long logs do get cut off :/

![image](https://github.com/woodpecker-ci/woodpecker/assets/24977596/4c8549d1-eb08-429a-b452-92dd3919e3a8)

**Reply by lafriks** [reply]
I think this is intentional to not make UI two slow to try to render large log (you can still download all logs using download button).

Improvement could be to show Load more button when logs are cut off to be able to load them in UI

**Reply by xoxys** [reply]
Would also like to see a button like this one:

![image](https://github.com/woodpecker-ci/woodpecker/assets/3391958/dec79844-68a1-4421-8b81-7374569ac587)

**Reply by qwerty287** [reply]
There's a separate issue for this https://github.com/woodpecker-ci/woodpecker/issues/2653 (I'll mark this comment as duplicate)

**Comment by 6543** [comment]
log view scroll bar is not wide enough to be easily clicked by mouse

**Comment by qwerty287** [comment]
Pagination does not work if you can't scroll down but there are more pages to load (especially important for e.g. secrets https://github.com/woodpecker-ci/woodpecker/pull/2873)

**Comment by qwerty287** [comment]
Remove the `Review changes` button, shouldn't be needed anymore (#1845 fixes the same underlying issue)

**Comment by qwerty287** [comment]
Sometimes (probably on very long branch names), icons have wrong size.

![grafik](https://github.com/woodpecker-ci/woodpecker/assets/80460567/b037bfd2-364f-4674-8e77-67e13a49cc9a)

**Comment by zc-devs** [comment]
Difficult to match Agent in Agents and Queue
![Agents](https://github.com/woodpecker-ci/woodpecker/assets/127358482/60fa69ef-cf59-4d73-a01d-3cc23762fc11)
![Queue](https://github.com/woodpecker-ci/woodpecker/assets/127358482/e2461656-8769-445a-ab6d-2fa6f87ebe65)

**Reply by anbraten** [reply]
similar to https://github.com/woodpecker-ci/woodpecker/issues/2948

**Reply by 6543** [reply]
Well for agents we already have all the info and could make the agentid in queue view a link to the agent

**Reply by lafriks** [reply]
Or better show agent name instead of id in task list

**Reply by qwerty287** [reply]
see https://github.com/woodpecker-ci/woodpecker/issues/5469

**Comment by qwerty287** [comment]
spinner looks ugly for > 9 running pipelines
![grafik](https://github.com/woodpecker-ci/woodpecker/assets/80460567/c85b981d-8e96-44d8-9d36-7c99bea15e4d)

**Reply by xoxys** [reply]
https://github.com/woodpecker-ci/woodpecker/pull/4379

**Comment by anbraten** [comment]
![image](https://github.com/woodpecker-ci/woodpecker/assets/6918444/ec5c4798-c1b5-4725-8381-2d1a1c59e6c1)

**Reply by xoxys** [reply]
Can you add a short description, what's wrong in the screenshot?

**Reply by anbraten** [reply]
Not sure what the original reason was. The long branch name looks weird and the icons should be in one column.

**Comment by qwerty287** [comment]
There are multiple similar issues here, but not that specific one:

Having long branches in the list results in different heights of the single items.

![grafik](https://github.com/woodpecker-ci/woodpecker/assets/80460567/f7857e04-8742-405b-9c19-f80f2d8c3d7f)

**Comment by zc-devs** [comment]
![Screenshot 2024-04-25 105522](https://github.com/woodpecker-ci/woodpecker/assets/127358482/6a0b9305-1b6f-4256-b6c0-1da694d8257c)

The timer resets after one hour

![Screenshot 2024-04-25 111830](https://github.com/woodpecker-ci/woodpecker/assets/127358482/cc4f552f-dac6-4188-9859-3f31e52b7c87)

`2.3.0`, but it was noticed in previous versions also.

**Comment by qwerty287** [comment]
Global secrets should also show up in org secrets list

**Comment by qwerty287** [comment]
The branches list behaves very weird sometimes.
1. items are duplicated
2. list is reloaded without interaction
3. items are even more than just duplicated

https://github.com/woodpecker-ci/woodpecker/assets/80460567/79190e0e-a9c8-4d63-a78d-4fffc1145bbc

@anbraten this somehow looks like a pagination issue

**Comment by qwerty287** [comment]
If you switch between pipelines (using the sidebar), the "changed files" number is not updated

**Comment by qwerty287** [comment]
If you directly try to open pipeline config like https://ci.woodpecker-ci.org/repos/3780/pipeline/17596/config it fails

**Comment by zc-devs** [comment]
Cron jobs are sorted by `update` time.

Create crons:
[Crons 1](https://github.com/woodpecker-ci/woodpecker/assets/127358482/e245a5fa-d634-4fe0-8694-658ecb43cb32)

Open `a` and just click `save`:
[Crons 2](https://github.com/woodpecker-ci/woodpecker/assets/127358482/8124af3f-3bd0-4f6c-9cd6-d8f91c9b4ff7)

**Comment by zc-devs** [comment]
Unknown error when branch doesn't exist:
[cron-branch-doesnt-exist](https://github.com/woodpecker-ci/woodpecker/assets/127358482/1e22c53c-2eb8-4d29-8a02-221252f2bf52)

Unknown error when wrong cron expression
[wrong-cron-expression](https://github.com/woodpecker-ci/woodpecker/assets/127358482/28cab314-e79a-478c-90ff-6330c9731aad)

Also there is one empty red error box.

**Comment by zc-devs** [comment]
There is empty `by` section for cron jobs
![Screenshot 2024-07-13](https://github.com/user-attachments/assets/efcf994a-9c80-440c-a4f8-0f2472574158)
#3898

**Reply by anbraten** [reply]
Instead of `pushed to triggered` we could just use `pushed to` 🤔 
![image](https://github.com/user-attachments/assets/1046a867-bf04-44cd-8991-9b02e6fe9ebc)


PS: awesome to have some testers of the `next` version directly reporting enhancements back ❤️

**Reply by anbraten** [reply]
We should use sth like a fixed width for the duration / since times to have the same width across pipeline items.

**Reply by zc-devs** [reply]
> fixed with...

For icon column too, then
![Screenshot 2](https://github.com/user-attachments/assets/b89cbe67-fefc-4830-bbe3-28da5a8665fc)

**Reply by zc-devs** [reply]
And this warning mark is annoying and useless to me. I have repos/branches/pipelines which always will be marked. Also, this mark pretty well visible from inside pipeline (`warnings` tab).

**Reply by zc-devs** [reply]
I've cleaned up a little bit, looks better to me:
[Without user](https://github.com/user-attachments/assets/a8a0fb5e-4918-4203-8137-0c0ba69d9f01)
[With user](https://github.com/user-attachments/assets/30c12740-5e05-4add-894b-1e2b4e6101a8)
Icons need improvements, it's just sketch.
PR number should have been the first:
[PR is first](https://github.com/user-attachments/assets/1e53ab57-c127-4db7-b43a-9971c837de21)

**Reply by qwerty287** [reply]
I think that looks better compared to now. Can you open a PR?

**Reply by zc-devs** [reply]
Nothing has been done in the code yet, I just played in the browser console. I can try to make a PR, but it will take some time.

**Reply by xoxys** [reply]
When has this tag-style look been added to the UI, and what was the motivation? Don't want to blame someone but this yellow rectangle... how could is haven been merged in this state at all? :see_no_evil: Sometimes it feels a bit of a step backwards.

**Reply by xoxys** [reply]
Ah, missed the PR link above. IMO, the new list style tries to add too much information now and the mobile view is close to unusable again.

**Reply by pat-s** [reply]
I would also be happy if we could fix/update/revert that new UI before releasing it.

**Comment by qwerty287** [comment]
cc @anbraten 

You can't login with the login button in the top right corner anymore. You're redirected to the start login page with an invalid oauth state error.

**Comment by 6543** [comment]
show message if logs where not found e.g. most probably deleted

example: https://ci.woodpecker-ci.org/repos/3780/pipeline/18869/19

![image](https://github.com/user-attachments/assets/c0be0fde-bb89-49f5-a72a-2935a1e3d388)

**Reply by qwerty287** [reply]
This can also just mean that a plugin didn't print anything

**Reply by 6543** [reply]
We should then have a way to mark logs as deleted

**Reply by anbraten** [reply]
We could just add a log line with something like: `Logs were deleted` if really necessary.

**Comment by qwerty287** [comment]
Org and global registries should show up in repo's/org's list

-> #5950

**Comment by qwerty287** [comment]
Disabled repos look pretty bad in the add repo list due to mixing `rounded-full` and `rounded-md`
![](https://github.com/user-attachments/assets/087233d5-fbb4-486c-9990-c9cb8a6d25d9)

**Reply by zc-devs** [reply]
But the first is a label and the last is a button...

**Reply by qwerty287** [reply]
I know, still it looks ugly to have these two types directly next to each other. It's better on every other list because we just use icon buttons there. Maybe this would be possible too there?
![grafik](https://github.com/user-attachments/assets/a60e41fd-7930-4b95-9451-53961851107e)

**Reply by pat-s** [reply]
I don't find it too bad, actually. I would try to align the overall heights of both first and see if this makes it more harmonic.

**Reply by zc-devs** [reply]
Was it introduced in `3.0`? Because on `2.7` [it looks very well](https://github.com/user-attachments/assets/6c8f0171-5923-4a45-963d-14e1efe3d281).
Maybe link the PR?

**Reply by qwerty287** [reply]
This only applies to disabled repos (repos that were active once already).

**Comment by wrenix** [comment]
Pre-filled variables in manual pipeline: https://github.com/woodpecker-ci/woodpecker/discussions/4404

Woodpecker:
![current state of woodpecker](https://github.com/user-attachments/assets/ae39e082-a602-431d-b428-d51edbff24ff)

Gitlab:
![on gitlab](https://about.gitlab.com/images/blogimages/Run-pipeline.gif)

**Comment by zc-devs** [comment]
1. Confusing `selected` sign for tabs in mobile view. Looks like closed list with some tasks. I've just tapped a couple of times and nothing happened :laughing: 
![Screenshot_20241123_1](https://github.com/user-attachments/assets/516061c0-9fd9-4dca-88f8-3855b8454225)
2. On mobile open the pipeline details. Open the step logs. Close the logs. Now there is no `selected` sign on Tasks tab.
![Screenshot_20241123_2](https://github.com/user-attachments/assets/902236f8-0fd8-49b0-aedb-8f2638b3a8f7)

**Reply by qwerty287** [reply]
Second issue is same on desktop. If you open a link to a step, the tasks tab is not selected. E.g. https://ci.woodpecker-ci.org/repos/3780/pipeline/23303/9

**Reply by xoxys** [reply]
https://github.com/woodpecker-ci/woodpecker/pull/4641

**Comment by zc-devs** [comment]
Icons and text are not aligned vertically:

![Screenshot 2024-11-28 1](https://github.com/user-attachments/assets/cfa24229-fcc7-44ff-a70a-2299b5315f00)

**Reply by zc-devs** [reply]
`next-b4055e8c10`, fixed. Can be closed/hidden.

**Comment by qwerty287** [comment]
On org settings, the link back to the org includes the space, which is different for other links.
![grafik](https://github.com/user-attachments/assets/78e08a8b-db37-42e2-88cb-82bfac3aac16)

**Reply by xoxys** [reply]
https://github.com/woodpecker-ci/woodpecker/pull/4497

**Comment by qwerty287** [comment]
Probably a regression of https://github.com/woodpecker-ci/woodpecker/pull/4403: The badge should be on the right side of the title, not below.
![grafik](https://github.com/user-attachments/assets/40d1956a-2637-4852-9732-378d5d2d597f)

**Reply by xoxys** [reply]
https://github.com/woodpecker-ci/woodpecker/pull/4499

**Comment by zc-devs** [comment]
#4458

Some icons are used not only with text
![Screenshot 2024-12-03 1](https://github.com/user-attachments/assets/c7348886-9c1c-4d25-bda3-486c4ae2e786)

and some, probably, aren't used with text at all
![Screenshot 2024-12-03 205327](https://github.com/user-attachments/assets/ad711573-b380-478c-a212-7a6750d6631c)

**Reply by xoxys** [reply]
And what's the point? The screenshots look good to me.

**Reply by zc-devs** [reply]
1. The icons can be used in multiple places => just reducing the size doesn't always work
![Screenshot 2024-12-03 3](https://github.com/user-attachments/assets/4e9a729f-7444-4bc5-89c5-dc769c4c63df)
2. Probably, it doesn't make sense to reduce size of _the buttons_, because there is no text right after them.

**Reply by xoxys** [reply]
I still don't get it. This is completely unrelated to the linked PR... Yes we changed the default icon size however the spacing etc especially in your screenshot have not even fitted before.

**Reply by zc-devs** [reply]
> This is completely unrelated to the linked PR

Not completely :)

1. Yes, the clock icon didn't fit as avatar. But now it fits even less.
2. What was wrong with _the buttons_? Where is _the text after them_? Where they did not fit and how?
![Screenshot 2024-12-03 222739](https://github.com/user-attachments/assets/476d3a2a-a13f-41cf-b498-644b33f8acfb)

**Reply by xoxys** [reply]
> Not completely :)

Not at all.

Nothing was wrong with the buttons... If we want them to be bigger again (personally I dont see why), we can change the icon size for the buttons. However, just adding what you would like to see instead of some meta issues would help.

**Reply by xoxys** [reply]
> Not completely :)

Not at all.

Nothing was wrong with the buttons... If we want them to be bigger again (personally I dont see why), we can change the icon size for the buttons. However, just adding what you would like to see instead of some meta issues would help.

**Reply by zc-devs** [reply]
1. I want to see cron job avatar the same size as user one. In pipeline list as well as in pipeline details.
2. I would like not to see the changes like `sed s/24/20/g` (without a reason).

PS: the icons with a text look better => get my ❤️ for that.

**Comment by pat-s** [comment]
When changing the default font-size in the browser, icons don't align as their size is hard-coded.

Font size 13 instead of 16 (default) in FF:

<img width="318" alt="image" src="https://github.com/user-attachments/assets/6f86ba6c-4703-403b-8930-04df3bc6c187" />

Solution would be to set their size relatively, i.e. use `1em` instead of pixels. 

`1em` example:

<img width="152" alt="image" src="https://github.com/user-attachments/assets/16c0ab3b-b2f7-4bd0-91e1-d31da0725748" />

**Reply by pat-s** [reply]
fixed by https://github.com/woodpecker-ci/woodpecker/pull/4575

**Comment by qwerty287** [comment]
The logs and step list are not moved to the background if you try to open the deploy popup. Probably a regression of the tailwind migration

![Screenshot 2025-01-12 at 09-07-03 Woodpecker](https://github.com/user-attachments/assets/a07e1911-09b6-4c1a-a026-79e60f277d88)

**Reply by xoxys** [reply]
Good catch. I can take care of this one.

**Reply by xoxys** [reply]
https://github.com/woodpecker-ci/woodpecker/pull/4715

**Comment by zc-devs** [comment]
Red color is oversaturated on OLED displays (to me).

**Reply by anbraten** [reply]
Same for me without oled display

**Reply by anbraten** [reply]
Contrast on dark mode is kinda broken.

**Comment by anbraten** [comment]
Buttons like those from the step list are missing `cursor-pointer` now.

**Reply by xoxys** [reply]
https://github.com/woodpecker-ci/woodpecker/pull/4982

**Comment by nykula** [comment]
Had to [redo almost completely](https://github.com/woodpecker-ci/woodpecker/blob/759b723a18a868b73c030400f19d59f725f08985/web/src/assets/locales/uk.json) the Ukrainian translation. I think the original author's intentions were good, but they mistranslated every term related to Git and CI. In the context of containers and pipelines, it should be like this:

- pipeline - конвеєр (term from IT), not трубопровід/газопровід (term from plumbing)
- event - подія (term from programming), not захід (term from concerts)
- pull request - запит на злиття (merge/pull), not запит на вилучення (deletion)
- push - вивантаження ("a commit push"), not натисни ("you! push a button")
- loading - завантаження (Ukrainian), not загрузка (Russian)
- branch - гілка (git), not відгалуження (branch of a pipe) or філія/відділення (branch of network, e.g. retail)
- Markdown - Markdown-розмітка (markup type), not уцінка (term from marketing)
- general - загальне (settings category), not генерал (military rank)

It might be worth mentioning in release notes for the next minor version that the translation has been fixed, maybe ask other Ukrainian users who might have considered the project buggy before to review it.

**Reply by 6543** [reply]
thanks!

If you like a can elevate your priviledges so you can review ukrainian translation and enhance the glasary ...

**Reply by nykula** [reply]
Thank you! I'll fill in the glossary tomorrow.

**Comment by toothstone** [comment]
Jobs give no information which agent ran them, or I can't find it. For more complex agent setups this is important debugging info.

**Reply by qwerty287** [reply]
I'm quite sure that there was an issue for this already. I can't find it though
